### PR TITLE
feat: Authentication

### PR DESCRIPTION
**Pull request by CarlosCRG19**
## Description

This implements the logic for authentication with a JWT as well as the signup/login endpoints

It solves #9, #10

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply_

- [x] Code compiles/runs correctly
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/bootstrap.py [hunk 1]
```diff
@@ -4,15 +4,30 @@
 from dotenv import load_dotenv
 
 from app.application.service import IBMDashboardService
-from app.infrastructure.object_storage.google_cloud_object_storage import GoogleCloudStorage
+from app.infrastructure import GoogleCloudStorage, PostgreSQLUserRepository,\
+    BcryptEncrypter, JWTManager
+
+JWT_ALGORITHM = "HS256"
+JWT_SECRET_KEY = "secret"
```
Comment (CristianCazares) [inline app/bootstrap.py]: Is the JWT_SECRET_KEY intended to the repo as public?

**Review by CristianCazares — APPROVED** [review]
Nice to see all the error codes organized!
I left a coment on `app/bootstrap.py`.

**Review by BrandonMagana — APPROVED** [review]
Interesting error handling amiga

File: app/bootstrap.py (hunk 1 above)
Comment (BrandonMagana) [reply app/bootstrap.py]: Already added those as env variables

**Pull request merged**